sys/include/event/periodic.h

```diff
@@ -75,14 +79,41 @@ void event_periodic_init(event_periodic_t *event_periodic, ztimer_clock_t *clock
  *
  * @param[in]   event_periodic   event_timout context object to use
  * @param[in]   interval         period length for the event
+ * @param[in]   count            time the event should repeat itself,
+ *                               EVENT_PERIODIC_FOREVER or 0 to run for ever,
+ *
+ * @pre count > 0
  */
-static inline void event_periodic_start(event_periodic_t *event_periodic,
-                                        uint32_t interval)
+static inline void event_periodic_start_iter(event_periodic_t *event_periodic,
```

Comment:
can we split this? keep the old `event_periodic_start()` doing `EVENT_PERIODIC_FOREVER`, add another that allows specifying the count?

Reply:
or, two init()? or, make init go forever and add another `event_periodic_set_maxcount()`?

The reason would be that the count thing is optional, but as is, the start function is overloaded.

Reply:
In the last commit, I had split them into two, `event_periodic_set_maxcount` sounds good as well. I don't have a preference.